scalableminds/webknossos
# Add segment groups

### URL of deployed dev instance (used for testing):
- https://segmentgroups.webknossos.xyz

### Steps to test:
- create a new annotation
- switch to the segments tab
- create some segments by brushing with some ids or simply click on existing segments
- right click on the root node in the segments tab to create one or multiple groups
- drag segments into the groups
- delete groups
- use undo/redo
- save & reload to check that everything was saved correctly

### Todo
- Front end (@philippotto)
  - [X] create groups
  - [X] delete group (recursively)
  - [X] renaming of groups
  - [X] dnd for groups and segments
  - [x] undo/redo for group mutations
    - [x] segments and groups can be moved higher than the root which makes them disappear
    - [x] groups are not shown when no segments exist at all
    - [x] undo might still be buggy. I fixed some bugs, but I still managed to break it at some point. I didn't find a clear reproduction, though.
    - [X] item is moved to wrong group sometimes
- [X] doublecheck export/import via NML (backend does this, for frontend, there is only a skeleton-related download, so no need to do adapt this in my opinion)
- ~~Handle merging of groups~~
- Fix e2e test

### Issues:
- fixes #6954 

------
(Please delete unneeded items, merge only when none are left open)
